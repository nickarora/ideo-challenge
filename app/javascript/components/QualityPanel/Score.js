import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export const Container = styled.section`
  font-size: 46px;
  font-weight: bold;
  margin-bottom: 48px;
`

export const Title = styled.small`
  display: block;
  font-size: 12px;
`

const Score = ({ quality }) => (
  <Container>
    <Title>your score:</Title>
    {quality.score}
  </Container>
)

Score.propTypes = {
  quality: PropTypes.shape({
    score: PropTypes.number.isRequired,
  }).isRequired,
}

export default Score
