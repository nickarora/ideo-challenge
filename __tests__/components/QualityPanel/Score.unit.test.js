import Score from '~/components/QualityPanel/Score'

describe('Score', () => {
  it('renders the given score', () => {
    const wrapper = shallow(
      <Score
        quality={{
          score: -25,
        }}
      >
        CONTENT
      </Score>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
