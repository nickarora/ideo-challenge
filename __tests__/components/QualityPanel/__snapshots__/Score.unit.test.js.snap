// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders the given score 1`] = `
<styled.section>
  <styled.small>
    your score:
  </styled.small>
  -25
</styled.section>
`;
